lib/workers/repository/update/branch/auto-replace.spec.ts

```diff
@@ -182,4 +182,706 @@ describe('workers/repository/update/branch/auto-replace', () => {
+      upgrade.replaceString = 'bitnami/redis:6.0.8';
+      upgrade.packageFile = 'Dockerfile';
+      upgrade.depIndex = 0;
+      upgrade.currentValue = '6.0.8';
+      const res = await doAutoReplace(upgrade, dockerfile, reuseExistingBranch);
+      expect(res).toBe(dockerfile.replace(upgrade.depName, upgrade.newName));
+    });
+
+    it('handles already replaced', async () => {
+      const dockerfile = 'FROM library/ubuntu:20.04';
+      upgrade.manager = 'dockerfile';
+      upgrade.updateType = 'replacement';
+      upgrade.depName = 'library/alpine';
+      upgrade.newName = 'library/ubuntu';
+      upgrade.packageFile = 'Dockerfile';
+      const res = await doAutoReplace(upgrade, dockerfile, reuseExistingBranch);
+      expect(res).toBe(dockerfile);
+    });
+
+    it('updates with terraform replacement', async () => {
+      const hcl =
+        'module "foo" {\nsource = "github.com/hashicorp/example?ref=v1.0.0"\n}';
+      upgrade.manager = 'terraform';
+      upgrade.updateType = 'replacement';
+      upgrade.depName = 'github.com/hashicorp/example';
+      upgrade.newName = 'github.com/hashicorp/new-example';
+      upgrade.currentValue = 'v1.0.0';
+      upgrade.depIndex = 0;
+      upgrade.packageFile = 'modules.tf';
+      const res = await doAutoReplace(upgrade, hcl, reuseExistingBranch);
+      expect(res).toBe(hcl.replace(upgrade.depName, upgrade.newName));
+    });
+
+    it('works with old name in depname', async () => {
+      const yml =
+        'image: "1111111111.dkr.ecr.us-east-1.amazonaws.com/my-repository:1"\n\n';
+      upgrade.manager = 'regex';
+      upgrade.updateType = 'replacement';
+      upgrade.depName =
+        '1111111111.dkr.ecr.us-east-1.amazonaws.com/my-repository';
+      upgrade.currentValue = '1';
+      upgrade.newName =
+        '1111111111.dkr.ecr.us-east-1.amazonaws.com/my-repository';
+      upgrade.depIndex = 0;
+      upgrade.replaceString =
+        'image: "1111111111.dkr.ecr.us-east-1.amazonaws.com/my-repository:1"\n\n';
+      upgrade.packageFile = 'k8s/base/defaults.yaml';
+      upgrade.matchStrings = [
+        'image:\\s*\\\'?\\"?(?<depName>[^:]+):(?<currentValue>[^\\s\\\'\\"]+)\\\'?\\"?\\s*',
+      ];
+      const res = await doAutoReplace(upgrade, yml, reuseExistingBranch);
+      expect(res).toBe(yml);
+    });
+
+    it('rebases if the deps list has changed', async () => {
+      upgrade.baseDeps = extractPackageFile(sampleHtml)?.deps;
+      reuseExistingBranch = true;
+      const res = await doAutoReplace(
+        upgrade,
+        'existing content',
+        reuseExistingBranch
+      );
+      expect(res).toBeNull();
+    });
+  });
+
+  describe('doReplacementAutoReplaceNewDepName', () => {
+    let reuseExistingBranch: boolean;
+    let upgrade: BranchUpgradeConfig;
+
+    beforeEach(() => {
+      upgrade = {
+        ...JSON.parse(JSON.stringify(defaultConfig)),
+      };
+      reuseExistingBranch = false;
+    });
```

Comment:
shift a level up and reuse for all tests

Reply:
done

Reply:
please read our contribution guide[^1]

[^1]: https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#resolve-review-comments-instead-of-commenting